crates/swc/src/dropped_comments_preserver.rs

```diff
@@ -100,16 +100,14 @@ impl DroppedCommentsPreserver {
     fn shift_leading_comments(&self, comments: &SingleThreadedComments) -> CommentEntries {
         let mut existing_comments = self.collect_existing_comments(comments);
 
-        for span in self.known_spans.iter() {
-            let (comments_to_move, next_byte_positions): (CommentEntries, CommentEntries) =
-                existing_comments
-                    .drain(..)
-                    .partition(|(bp, _)| *bp <= span.lo);
-
-            existing_comments.extend(next_byte_positions);
+        existing_comments.sort_by(|(bp_a, _), (bp_b, _)| bp_a.cmp(bp_b));
```

Comment:
I have noticed that sorting occurs in the `﻿collect_existing_comments`. This function is primarily intended to ensure that the ﻿comments are sorted, guarding against any possible changes within ﻿`collect_existing_comments`.